feat(annotations): add vertica.com/disable-extra-paths-auto-mount

matchaxnb opened the pull request:
This complements the disable-depot-volume-management annotation.

When a database is revived, the extraPaths annotation get automatically populated from the metadata. This has negative effects when volumes are managed manually such as when they're self-managed. In our case, the temp volume is on a local fast SSD, and our config ("local-data" in Vertica parlance) is on a RBD to allow for rescheduling. We don't want temp storage on RBD, so we don't want to get our TEMP storage location force-mounted as the RBD.

Using this annotation allows to skip the extraPaths auto-mounting thus disabling this detrimental feature in an explicit way.

# How to use 

Set the following annotation in your VerticaDB object:

`"vertica.com/disable-extra-paths-auto-mount": "true"`

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/vertica/vertica-kubernetes?pullRequest=1177) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/vertica/vertica-kubernetes?pullRequest=1177) before we can accept your contribution.<br/><hr/>**Matcha (from Criteo)** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/vertica/vertica-kubernetes?pullRequest=1177) it.</sub>

matchaxnb committed:
feat(annotations): add vertica.com/disable-extra-paths-auto-mount

This complements the disable-depot-volume-management annotation.

When a database is revived, the extraPaths annotation get automatically
populated from the metadata. This has negative effects when volumes are
managed manually such as when they're self-managed. In our case, the
temp volume is on a local fast SSD, and our config ("local-data" in
Vertica parlance) is on a RBD to allow for rescheduling. We don't want
temp storage on RBD, so we don't want to get our TEMP storage location
force-mounted as the RBD.

Using this annotation allows to skip the extraPaths auto-mounting thus
disabling this detrimental feature in an explicit way.

roypaulin commented:
Looks good! You can't run testsfrom a fork branch so I will merge this first to a feature branch in the vertica account.
